### PR TITLE
Moving the constant parameter check to a more common file

### PR DESCRIPTION
**Pull request by Gamrix**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #63198
* __->__ #64251

Differential Revision: [D31161850](https://our.internmc.facebook.com/intern/diff/D31161850)

### CODE REVIEW

File: torch/csrc/jit/passes/frozen_graph_optimizations.h [hunk 1]
```diff
@@ -13,6 +13,10 @@
 namespace torch {
 namespace jit {
 
+bool nonConstantParameters(Node* n);
+// Checks if the parameters, not including the
+// first param are all constants.
```
Comment (eellison) [inline torch/csrc/jit/passes/frozen_graph_optimizations.h:18]: nit: typically you add the comments above the API, not below

File: torch/csrc/jit/passes/frozen_graph_optimizations.h [hunk 2]
```diff
@@ -13,6 +13,10 @@
 namespace torch {
 namespace jit {
 
+bool nonConstantParameters(Node* n);
```
Comment (eellison) [inline torch/csrc/jit/passes/frozen_graph_optimizations.h:16]: nit: could we move this to something like `torch/csrc/jit/passes/optimization_utils` ?

File: torch/csrc/jit/passes/frozen_graph_optimizations.cpp [hunk 3]
```diff
@@ -22,6 +38,7 @@ void OptimizeFrozenGraph(
       FoldFrozenConvAddOrSub(graph);
       FoldFrozenConvMulOrDiv(graph);
     }
+    FrozenConcatLinear(graph);
```
Comment (eellison) [inline torch/csrc/jit/passes/frozen_graph_optimizations.cpp:41]: i think you meant this to be in the other pr

File: torch/csrc/jit/passes/frozen_graph_optimizations.cpp [hunk 4]
```diff
@@ -1,15 +1,31 @@
 #include <c10/util/irange.h>
 #include <torch/csrc/jit/ir/alias_analysis.h>
+#include <torch/csrc/jit/ir/ir.h>
 #include <torch/csrc/jit/ir/ir_views.h>
+#include <torch/csrc/jit/passes/frozen_concat_linear.h>
```
Comment (eellison) [inline torch/csrc/jit/passes/frozen_graph_optimizations.cpp:5]: same here..

File: torch/csrc/jit/passes/frozen_graph_optimizations.cpp [hunk 5]
```diff
@@ -1,15 +1,31 @@
...
 #include <torch/csrc/jit/passes/frozen_conv_folding.h>
 #include <torch/csrc/jit/passes/frozen_graph_optimizations.h>
 #include <torch/csrc/jit/passes/remove_dropout.h>
 #include <torch/csrc/jit/runtime/graph_executor.h>
 #include <torch/csrc/utils/memory.h>
+#include <vector>
+#include "ATen/core/functional.h"
```
Comment (eellison) [inline torch/csrc/jit/passes/frozen_graph_optimizations.cpp:12]: "ATen/core/functional.h" will break some internal builds -> `<ATen/core/functional.h>`

**Comment by Gamrix** [comment]
Ugh, didn't pay quite enough attention to the git interactive rebase.

**Review by eellison — APPROVED** [review]

**Comment by Gamrix** [comment]
@gamrix has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31161850).

**Pull request closed without merging**